File: Samples for Solid Edge/[C#]-Samples for Solid Edge/C#/SolidEdge.UnitsOfMeasure/InternalExample.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;

namespace SolidEdge.UnitsOfMeasure
{
    public class InternalExample
    {
        private SolidEdgeFramework.UnitTypeConstants _unitType;
        private SolidEdgeConstants.PrecisionConstants _precision;
        private object _externalValue;
        private double _internalValue = 0.0254;

        public InternalExample()
        {
            _unitType = SolidEdgeFramework.UnitTypeConstants.igUnitDistance;
            _precision = SolidEdgeConstants.PrecisionConstants.igPrecisionThousandths;
            UpdateExternalValue();
        }

        public SolidEdgeFramework.UnitTypeConstants UnitType
        {
            get
            {
                return _unitType;
            }
            set
            {
                _unitType = value;
                UpdateExternalValue();
            }
        }

        public SolidEdgeConstants.PrecisionConstants Precision
        {
            get
            {
                return _precision;
            }
            set
            {
                _precision = value;
                UpdateExternalValue();
            }
        }

        [DescriptionAttribute("Value in user display units.")]
        public object ExternalValue
        {
            get
            {
                return _externalValue;
            }
        }

        [DescriptionAttribute("Value in internal (database) units.")]
        public double InternalValue
        {
            get
            {
                return _internalValue;
            }
            set
            {
                _internalValue = value;
                UpdateExternalValue();
            }
        }

        private void UpdateExternalValue()
        {
            SolidEdgeFramework.Application application = null;
            SolidEdgeFramework.SolidEdgeDocument document = null;
            SolidEdgeFramework.UnitsOfMeasure unitsOfMeasure = null;

            try
            {
                application = (SolidEdgeFramework.Application)Marshal.GetActiveObject("SolidEdge.Application");
                document = (SolidEdgeFramework.SolidEdgeDocument)application.ActiveDocument;
                unitsOfMeasure = document.UnitsOfMeasure;
                _externalValue = unitsOfMeasure.FormatUnit((int)_unitType, _internalValue, _precision);
            }
            catch (System.Exception ex)
            {
                _externalValue = ex.Message;
            }
            finally
            {
                if (unitsOfMeasure != null) Marshal.ReleaseComObject(unitsOfMeasure);
                if (document != null) Marshal.ReleaseComObject(document);
                if (application != null) Marshal.ReleaseComObject(application);
            }
        }
    }
}

File: Samples for Solid Edge/[VB]-Samples for Solid Edge/VB/SolidEdge.UnitsOfMeasure/InternalExample.vb
Imports System.ComponentModel
Imports System.Runtime.InteropServices
Imports System.Text

Namespace SolidEdge.UnitsOfMeasure
	Public Class InternalExample
		Private _unitType As SolidEdgeFramework.UnitTypeConstants
		Private _precision As SolidEdgeConstants.PrecisionConstants
		Private _externalValue As Object
		Private _internalValue As Double = 0.0254

		Public Sub New()
			_unitType = SolidEdgeFramework.UnitTypeConstants.igUnitDistance
			_precision = SolidEdgeConstants.PrecisionConstants.igPrecisionThousandths
			UpdateExternalValue()
		End Sub

		Public Property UnitType() As SolidEdgeFramework.UnitTypeConstants
			Get
				Return _unitType
			End Get
			Set(ByVal value As SolidEdgeFramework.UnitTypeConstants)
				_unitType = value
				UpdateExternalValue()
			End Set
		End Property

		Public Property Precision() As SolidEdgeConstants.PrecisionConstants
			Get
				Return _precision
			End Get
			Set(ByVal value As SolidEdgeConstants.PrecisionConstants)
				_precision = value
				UpdateExternalValue()
			End Set
		End Property

		<DescriptionAttribute("Value in user display units.")> _
		Public ReadOnly Property ExternalValue() As Object
			Get
				Return _externalValue
			End Get
		End Property

		<DescriptionAttribute("Value in internal (database) units.")> _
		Public Property InternalValue() As Double
			Get
				Return _internalValue
			End Get
			Set(ByVal value As Double)
				_internalValue = value
				UpdateExternalValue()
			End Set
		End Property

		Private Sub UpdateExternalValue()
			Dim application As SolidEdgeFramework.Application = Nothing
			Dim document As SolidEdgeFramework.SolidEdgeDocument = Nothing
			Dim unitsOfMeasure As SolidEdgeFramework.UnitsOfMeasure = Nothing

			Try
				application = CType(Marshal.GetActiveObject("SolidEdge.Application"), SolidEdgeFramework.Application)
				document = CType(application.ActiveDocument, SolidEdgeFramework.SolidEdgeDocument)
				unitsOfMeasure = document.UnitsOfMeasure
				_externalValue = unitsOfMeasure.FormatUnit(CInt(_unitType), _internalValue, _precision)
			Catch ex As System.Exception
				_externalValue = ex.Message
			Finally
				If unitsOfMeasure IsNot Nothing Then
					Marshal.ReleaseComObject(unitsOfMeasure)
				End If
				If document IsNot Nothing Then
					Marshal.ReleaseComObject(document)
				End If
				If application IsNot Nothing Then
					Marshal.ReleaseComObject(application)
				End If
			End Try
		End Sub
	End Class
End Namespace
